dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java

```diff
@@ -1073,7 +1072,8 @@ public Result<Object> delete(User loginUser, String fullName,
         if (resourcesNeedToDeleteSet.size() > 0) {
             for (ResourcesTask resourcesTask : resourcesNeedToDeleteSet) {
                 int taskId = resourcesTask.getTaskId();
-                if (processService.isTaskOnline(taskDefinitionMapper.selectById(taskId).getCode())) {
+                TaskDefinition taskDefinition = taskDefinitionMapper.selectById(taskId);
+                if (taskDefinition != null && processService.isTaskOnline(taskDefinition.getCode())) {
```

Comment:
According to our earlier discussion, we don't need to check its online status, https://github.com/apache/dolphinscheduler/issues/13596

Reply:
You mean the restriction on deleting has no meaning because users could directly delete files which is stored in third-party storage system. So here we do not add check just let the users go ahead. When users invoke task, they would find the referenced resource file not existing error log. I think I got your point and I modify my PR again. Thanks for your remind!